test/values.test.js

```diff
@@ -501,12 +538,228 @@ describe('Values', () => {
+      }),
+      'Query(Select(["favorites", "foods", 1], {favorites: {foods: ["crunchings", "munchings", "lunchings"]}}))'
+    )
+
+    // Map an Array
+    assertPrint(
+      new Query({
+        map: { lambda: 'x', expr: { add: [{ var: 'x' }, 1] } },
+        collection: [1, 2, 3],
+      }),
+      'Query(Map([1, 2, 3], Lambda("x", Add(Var("x"), 1))))'
+    )
+
+    // Map a Page
+    new assertPrint(
+      new Query({
+        map: {
+          expr: { get: { var: 'x' } },
+          lambda: 'x',
+        },
+        collection: { paginate: { databases: null } },
+      }),
+      'Query(Map(Paginate(Databases()), Lambda("x", Get(Var("x")))))'
+    )
+
+    // Foreach
+    assertPrint(
+      new Query({
+        collection: { paginate: { documents: { collection: 'orders' } } },
+        foreach: {
+          lambda: 'doc',
+          expr: {
+            update: { var: 'doc' },
+            params: { object: { data: { object: { createdBy: 'Ryan' } } } },
+          },
+        },
+      }),
+      `Query(Foreach(Paginate(Documents(Collection("orders"))), Lambda("doc", Update(Var("doc"), {data: {createdBy: "Ryan"}}))))`
+    )
+
+    // If
+    assertPrint(
+      new Query({
+        else: 'was false',
+        then: 'was true',
+        if: true,
+      }),
+      `Query(If(true, "was true", "was false"))`
+    )
+
+    // Call
+    assertPrint(
+      new Query({
+        arguments: 2,
+        call: {
+          function: 'double',
+        },
+      }),
+      `Query(Call(Function("double"), 2))`
+    )
+
+    assertPrint(
+      new Query({
+        arguments: [1, 2, 3],
+        call: {
+          function: 'double',
+        },
+      }),
+      `Query(Call(Function("double"), [1, 2, 3]))`
+    )
+
+    assertPrint(
+      new Query({
+        arguments: {
+          object: {
+            one: 1,
+            two: 2,
+            three: 3,
+          },
+        },
+        call: {
+          function: 'double',
+        },
+      }),
+      `Query(Call(Function("double"), {one: 1, two: 2, three: 3}))`
+    )
+
+    // Databases
+    assertPrint(
+      new Query({
+        databases: {
+          database: 'lettuce',
+        },
+      }),
+      'Query(Databases(Database("lettuce")))'
+    )
+
+    assertPrint(
+      new Query({
+        databases: null,
+      }),
+      'Query(Databases())'
+    )
+
+    // Collections
+    assertPrint(
+      new Query({
+        collections: {
+          database: 'lettuce',
+        },
+      }),
+      'Query(Collections(Database("lettuce")))'
+    )
+
+    assertPrint(
+      new Query({
+        collections: null,
+      }),
+      'Query(Collections())'
+    )
+
+    // Documents
+    assertPrint(
+      new Query({
+        documents: {
+          collection: 'lettuce',
+        },
+      }),
+      'Query(Documents(Collection("lettuce")))'
+    )
+
+    assertPrint(
+      new Query({
+        documents: null,
+      }),
+      'Query(Documents())'
```

Comment:
that assertion is not correct, if `documents` field is null, then it should render `Documents(null)`

Reply:
@marrony Ah, okay. I was wondering why that was failing 😅